pulp_container/app/models.py

```diff
@@ -154,40 +164,93 @@ def init_image_nature(self):
             # it suffices just to have a single manifest of a specific nature;
             # there is no case where the manifest is both bootable and flatpak-based
-            if manifest.is_bootable:
+            if manifest.type == MANIFEST_TYPE.BOOTABLE:
+                self.type = MANIFEST_TYPE.BOOTABLE
                 self.is_bootable = True
                 return True
-            elif manifest.is_flatpak:
+            elif manifest.type == MANIFEST_TYPE.FLATPAK:
+                self.type = MANIFEST_TYPE.FLATPAK
                 self.is_flatpak = True
                 return True
 
-        return False
+        return updated_type
 
     def init_manifest_nature(self):
         if self.is_bootable_image():
+            # DEPRECATED: is_bootable is deprecated and will be removed in a future release.
             self.is_bootable = True
+            self.type = MANIFEST_TYPE.BOOTABLE
             return True
         elif self.is_flatpak_image():
+            # DEPRECATED: is_flatpak is deprecated and will be removed in a future release.
             self.is_flatpak = True
+            self.type = MANIFEST_TYPE.FLATPAK
             return True
-        else:
-            return False
+        elif self.is_helm_chart():
+            self.type = MANIFEST_TYPE.HELM
+            return True
+        elif media_type := self.is_cosign():
+            self.type = self.get_cosign_type(media_type)
+            return True
+        elif self.is_manifest_image():
+            self.type = MANIFEST_TYPE.IMAGE
+            return True
+
+        return False
 
     def is_bootable_image(self):
-        if (
+        return (
             self.annotations.get("containers.bootc") == "1"
             or self.labels.get("containers.bootc") == "1"
-        ):
-            return True
-        else:
-            return False
+        )
 
     def is_flatpak_image(self):
         return True if self.labels.get("org.flatpak.ref") else False
 
+    def is_manifest_image(self):
+        return self.media_type in MANIFEST_MEDIA_TYPES.IMAGE
+
+    @property
+    def json_manifest(self):
+        return json.loads(self.data)
+
+    def is_cosign(self):
+        try:
+            # layers is not a mandatory field
+            layers = self.json_manifest["layers"]
+        except KeyError:
+            return False
+
+        for layer in layers:
+            if layer["mediaType"] in COSIGN_MEDIA_TYPES:
+                return layer["mediaType"]
+        return False
+
+    def get_cosign_type(self, media_type):
+        if media_type in MEDIA_TYPE.COSIGN_SBOM:
+            return MANIFEST_TYPE.COSIGN_SBOM
+
+        cosign_types_mapping = {
+            MEDIA_TYPE.COSIGN_BLOB: MANIFEST_TYPE.COSIGN_SIGNATURE,
+            MEDIA_TYPE.COSIGN_ATTESTATION: MANIFEST_TYPE.COSIGN_ATTESTATION,
+            MEDIA_TYPE.COSIGN_ATTESTATION_BUNDLE: MANIFEST_TYPE.COSIGN_ATTESTATION_BUNDLE,
+        }
+        return cosign_types_mapping.get(media_type)
```

Comment:
We should improve the readability aspect of this method. Is such a level of granularity even needed? If yes, let's move this mapping to `constants.py`.